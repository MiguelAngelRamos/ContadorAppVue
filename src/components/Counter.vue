<template>
  <h2>{{customTitle}}</h2>
  <p>{{contador }} <sup>2</sup> = {{ squareCounter }}</p>
  <p data-testid="counter">{{ contador }}</p>
  <div>
    <button class="btn btn-success mx-2" @click="incrementar">+1</button>
    <button class="btn btn-danger" @click="disminuir">-1</button>
  </div>
</template>

<script>
export default {
  name: 'Counter',
  props: {
    title: String,
    start: {
      type: Number,
      default: 100,
      validator(value) {
        // PARA EL EQUIPO TRABAJO DE PROGRAMADORES
        return value >=0;
      }
    }
  },
  data() {
    return {
      contador : this.start,
    }
  },
  methods: {
    incrementar() {
      console.log('aumentar contador');
      this.contador++;
    },
    disminuir() {
      console.log('disminuir contador');
      this.contador--;
    }
  },
  computed: {
    squareCounter() {
      return Math.pow(this.contador, 2);
    },
    customTitle() {
      return this.title || "Title Counter Default"
    }
  }

}
</script>

<style>

</style>