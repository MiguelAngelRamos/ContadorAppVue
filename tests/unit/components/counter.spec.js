import { shallowMount } from "@vue/test-utils";
import Counter from '@/components/Counter';

// CREAMOS NUESTRA SUITE
describe('Pruebas del componente Counter.vue', ()=> {

  test('Debemos hacer match con el snapshot', () => {
    // Nuestro objeto de prueba
    const wrapper = shallowMount(Counter);
    // console.log(wrapper.html());
    expect(wrapper.html()).toMatchSnapshot();
  })
});