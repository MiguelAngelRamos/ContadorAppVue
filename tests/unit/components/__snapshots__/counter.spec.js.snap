// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del componente Counter.vue Debemos hacer match con el snapshot 1`] = `
<h2>Title Counter Default</h2>
<p>Miguel</p>
<p>100 <sup>2</sup> = 10000</p>
<p>Vue js es un excelente framework</p>
<div><button class="btn btn-success mx-2">+1</button><button class="btn btn-danger">-1</button></div>
`;
